server/bin/pbench-index.py

```diff
@@ -45,6 +43,28 @@
 _filenotfounderror = getattr(__builtins__, "FileNotFoundError", IOError)
 
 
+class ErrorCode:
+    def __init__(self, name, value, tarball_error, message):
+        self.name = name
+        self.value = value
+        self.tarball_error = tarball_error
+        self.message = message
+
+
+class Errors:
+    def __init__(self, *codes):
+        self.errors = dict()
+        for code in codes:
+            self.errors[code.name] = code
+
+    def __getitem__(self, key):
+        return self.errors[key]
+
+
+def success(res):
+    return res.value == 0
```

Comment:
I would make this a method of `ErrorCode`, so that you can use `res.success()` ... it's a property of the Error, not an unrelated function.

Another possibility would be for `ErrorCode`'s `__init__` to create an instance property `success` like

```
    self.success = value == 0
```

and then you could just use `res.success`